frontend/src/main/java/com/google/tripmeout/frontend/storage/InMemoryTripModelStorage.java

```diff
@@ -0,0 +1,85 @@
+package com.google.tripmeout.frontend.storage;
+
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripAlreadyExistsException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class InMemoryTripModelStorage {
```

Comment:
Please make this implement `TripStorage`. Doing so will allow this to be plugged in wherever a `TripStorage` is required. It will also reveal a compile-time error due to the exception signature not matching that of the interface. (I'd recommend using `RuntimeException`s for these anyway, unless you've already talked about this with Chris).